plutus-ledger/src/Ledger/Generators.hs

```diff
@@ -195,20 +197,23 @@ genValidTransaction' g (Mockchain _ ops _) = do
     nUtxo <- if Map.null ops
                 then Gen.discard
                 else Gen.int (Range.linear 1 (Map.size ops))
-    let ins = pubKeyTxIn . fst <$> inUTXO
+    let ins = (`TxInputWitnessed` ConsumePublicKeyAddress) . fst <$> inUTXO
         inUTXO = take nUtxo $ Map.toList ops
         totalVal = foldl' (<>) mempty $ map (txOutValue . snd) inUTXO
     genValidTransactionSpending' g ins totalVal
 
 genValidTransactionSpending :: MonadGen m
-    => [TxIn]
+    => [TxInputWitnessed]
     -> Value
     -> m CardanoTx
 genValidTransactionSpending = genValidTransactionSpending' generatorModel
 
+-- | A transaction input, consisting of a transaction output reference and an input type with data witnesses.
+data TxInputWitnessed = TxInputWitnessed !TxOutRef !Ledger.TxInType
```

Comment:
What motivated this change?

Reply:
tldr: introduced as least friction fix

Wanted to leave implementation of `genValidTransactionSpending` that was using `TxIn`s. `txInType` in `TxIn` is optional, which made it impossible as its not optional in `TxInput`. The code can remain without big rewrite using `TxIn` without optionality, that is `TxInputWitnessed`.